bigframes/core/blocks.py

```
@@ -221,7 +221,14 @@ def shape(self) -> typing.Tuple[int, int]:
             except Exception:
                 pass
 
-        row_count = self.session._executor.get_row_count(self.expr)
+        row_count = next(
+            iter(
+                self.session._executor.execute(self.expr.row_count())
+                .to_arrow_table()
+                .to_pydict()
+                .values()
+            )
+        )[0]
```

Do we want to check and fail fast if the number of values is not equal to 1?

---

Actually pulled this in to the execute result object itself, and add asserts